rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -260,13 +260,26 @@ private boolean validateLogsBloom(BlockHeader header, BlockResult result) {
         return Arrays.equals(calculateLogsBloom(result.getTransactionReceipts()), header.getLogsBloom());
     }
 
+    @VisibleForTesting
+    public BlockResult executeForMining(Block block, BlockHeader parent, boolean discardInvalidTxs) {
```

Comment:
This method is unnecessary as it is only used in tests. Can be replaced by a call to the other executeForMining.

Reply:
Indeed, it's weird why it is like this in master `@VisibleForTest` and Public together make no sense